integrations/gtag/lib/index.js

```diff
@@ -0,0 +1,893 @@
+};
+
+/**
+ * User signed Up - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.signedUpEnhanced = function(track) {
+  var props = track.properties();
+  push('event', 'sign_up', { method: props.method });
+};
+
+/**
+ * Lead Generated - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.leadGeneratedEnhanced = function(track) {
+  push('event', 'generate_lead', {
+    transaction_id: track.id(),
+    value: track.price(),
+    currency: track.currency()
+  });
+};
+
+/**
+ * Exception - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.exceptionOccuredEnhanced = function(track) {
+  var props = track.properties();
+  push('event', 'exception', {
+    description: props.description,
+    fatal: props.fatal
+  });
+};
+
+/**
+ * Timing Completed - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.timingCompletedEnhanced = function(track) {
+  push('event', 'timing_complete', {
+    name: track.name(),
+    value: track.value()
+  });
+};
+
+/**
+ * Track page view event.
+ *
+ * @api private
+ * @param { Page } page
+ * @param  opt
+ * @param  props
+ */
+
+function trackPageViewEvent(page, opts, props) {
+  var nonInteraction =
+    props && props.nonInteraction !== undefined
+      ? !!props.nonInteraction
+      : !!(opts && opts.nonInteraction);
+  push('event', 'page_view', {
+    page_title: page.title,
+    page_location: page.url,
+    page_path: page.path,
+    non_interaction: nonInteraction
+  });
+}
+
+/**
+ * Set custom dimensions and metrics.
+ *
+ * @api private
+ * @param options
+ */
+
+function setCustomDimensionsAndMetrics(options) {
+  var gaOptions = options.gaOptions || {};
+  if (gaOptions && Object.keys(gaOptions).length) {
+    if (gaOptions.setAllMappedProps) {
+      // set custom dimension and metrics if present
+      // REF: https://developers.google.com/analytics/devguides/collection/gtagjs/custom-dims-mets
+
+      var customMap = merge(gaOptions.dimensions, gaOptions.metrics);
+      if (options.gaWebMeasurementId) {
+        push('config', options.gaWebMeasurementId, {
+          custom_map: customMap
+        });
+      }
+      if (options.gaWebAppMeasurementId) {
+        push('config', options.gaWebAppMeasurementId, {
+          custom_map: customMap
+        });
+      }
+    }
+  }
+}
+
+/**
+ * Enhanced ecommerce format data for checkout.
+ *
+ * @api private
+ * @param {Track} track
+ */
+function extractCheckoutOptions(track) {
+  var props = track.properties();
+  var total = track.total() || track.revenue() || 0;
+  var coupon = track.proxy('properties.coupon');
+  var options = reject([
+    track.proxy('properties.paymentMethod'),
+    track.proxy('properties.shippingMethod')
+  ]);
+
+  return {
+    currency: track.currency(),
+    value: total,
+    items: getFormattedProductList(track),
+    coupon: coupon,
+    checkout_step: props.step || 1,
+    checkout_option: options.length ? options.join(', ') : null
```

Comment:
I think each step should only have 1 checkout_option? Ie. first checkout step is where a customer enters shipping information. So `checkout_step` would be `1` and `checkout_option` would be the shipping method. Then the next `checkout_step` would be `2` where a user enters bulling/payment information and then the `checkout_option` would be a users `paymentMethod`. Is there documentation you found where we would want to be sending both in this format you could share? I was having a hard time finding it.

Reply:
@briemcnally not really, it depends on the application using sdk, in some cases user can enter paymentMethod and shippingMethod on same page that can be tracked as single step.

Reply:
I agree with @vjnathe-webonise here, wtr to keeping up with parity.

Reply:
Sounds good! Let’s keep it the way it is implemented.